addons/web/static/src/js/widgets/colorpicker.js

```diff
@@ -276,6 +276,14 @@ var ColorpickerWidget = Widget.extend({
     _onClick: function (ev) {
         ev.originalEvent.__isColorpickerClick = true;
         $(ev.target).find('> .o_opacity_pointer, > .o_slider_pointer, > .o_picker_pointer').addBack('.o_opacity_pointer, .o_slider_pointer, .o_picker_pointer').focus();
+        
+        switch ($(ev.target).data('colorMethod')) {
```

Comment:
Avoid using jQuery's `.data` method when possible, it can create inconsistencies in the way the data is read or written because of how jQuery handles loading and writing the data to and from data-attributes.

Use `HTMLElement.dataset` instead

Since you're treating a single case, you can simply use an if instead of a switch-case also :)